File: C++/1961.cpp
#include<stdio.h>

int main(){

    int pulo = 0, qtd = 0, i = 0;

    scanf("%d %d", &pulo, &qtd);
    int torres[qtd];

    for(i = 0; i < qtd; i++){
        int aux;
        scanf("%d", &aux);
        torres[i] = aux;
    }

    int tam = qtd-2;
    int verifica = 1;
    int l = 0;

    while(l <= tam){
        if(torres[l+1] > torres[l]){
            if ((torres[l]+pulo) < torres[l+1])
                verifica = 0;
        }else{
            if ((torres[l+1]+pulo) < torres[l])
                verifica = 0;   
        }
    
        l++;
    }

    if(verifica == 1){
        puts("YOU WIN");
    }else{
        puts("GAME OVER");
    }

    return 0;
}

File: Python/falta gravar/1961.py
pulo, qtd = input().split()

pulo = int(pulo)
qtd = int(qtd)
torres = []

torres = list(map(int, input().split(' ')))

tam = len(torres)-2
verifica = 1
l = 0
while l <= tam:
    if torres[l+1] > torres[l]:
        if (torres[l]+pulo) < torres[l+1]:
            verifica = 0
    else:
        if (torres[l+1]+pulo) < torres[l]:
            verifica = 0

    l+=1


if verifica == 1:
    print('YOU WIN')
else:
    print('GAME OVER')
